packages/shared/src/utils.ts

```diff
@@ -41,13 +46,14 @@ export function getValues(arr: AnyNumArray): NumArray {
 }
 
 export function toTypedNdArray<T extends NumArray>(
-  arr: NdArray<T>
-): NdArray<Exclude<T, number[]> | Float32Array> {
+  arr: NdArray<T>,
+  TypedArrayConstructor: Float32ArrayConstructor | Uint8ArrayConstructor
```

Comment:
Passing the typed array constructor as mentioned. Unfortunately, there doesn't seem to be a union type of all possible typed array constructors...

Reply:
Unfortunate.

You could create the `TypedArrayConstructor` type on the same model of `TypedArray`:
```
    type TypedArrayConstructor =
        | Int8ArrayConstructor
        | Int16ArrayConstructor
        | Int32ArrayConstructor
        | Uint8ArrayConstructor
        | Uint8ClampedArrayConstructor
        | Uint16ArrayConstructor
        | Uint32ArrayConstructor
        | Float32ArrayConstructor
        | Float64ArrayConstructor;
```